well_known.go

```diff
@@ -0,0 +1,51 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+// WellKnownResult is the result of looking up a matrix server's well-known file.
+// Located at https://<server_name>/.well-known/matrix/server
+type WellKnownResult struct {
+	NewAddress ServerName `json:"m.server"`
+}
+
+// LookupWellKnown looks up a well-known record for a matrix server. If one if
+// found, it returns the server to redirect to.
+func LookupWellKnown(serverNameType ServerName) (*WellKnownResult, error) {
+	serverName := string(serverNameType)
+
+	// Handle ending "/"
+	strings.Trim(serverName, "/")
```

Comment:
surely it is an error for a server name to include a / ? we should reject it, not trim it.